docker/lookaside.go

```diff
@@ -45,11 +52,11 @@ type registryNamespace struct {
 }
 
 // signatureStorageBase is an "opaque" type representing a lookaside Docker signature storage.
-// Users outside of this file should use configuredSignatureStorageBase and signatureStorageURL below.
+// Users outside of this file should use ConfiguredSignatureStorageBase and signatureStorageURL below.
 type signatureStorageBase *url.URL // The only documented value is nil, meaning storage is not supported.
```

Comment:
This does not work for external callers if we don’t export `signatureStorageURL` (can the result even be used in Podman without an explicit `*url.URL` cast?)

… but there’s not _that much_ benefit to doing just that, because any caller would hard-code the semantics of the signature storage (in particular the sequential indices) anyway. We would need a very generic ReadSignatures/WriteSignatures/DeleteSignature API, which seems more effort than is strictly necessary right now.

So…

Reply:
the `dockerReference` also needs to be exported

Reply:
Exporting that would not directly help; users of `alltransports.ParseImageName` or `docker.NewReference` only get a generic `types.ImageReference` interface.

`GetRepositoryTags` already handles that by accepting a `types.ImageReference` and making a runtime check. That’s ugly but convenient enough for callers, and impossible to misuse.

Reply:
Thanks.  Add the same type assertion check inside `SignatureStorageBaseURL`